backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/tools/AcceptActionPayloadNodeToolProvider.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.diagram.common.view.tools;
+
+import java.util.Objects;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.sirius.components.view.builder.IViewDiagramElementFinder;
+import org.eclipse.sirius.components.view.builder.generated.DiagramBuilders;
+import org.eclipse.sirius.components.view.builder.generated.ViewBuilders;
+import org.eclipse.sirius.components.view.builder.providers.INodeToolProvider;
+import org.eclipse.sirius.components.view.diagram.NodeTool;
+import org.eclipse.syson.util.AQLConstants;
+import org.eclipse.syson.util.IDescriptionNameGenerator;
+
+/**
+ * Used to create element as a payload of an accept action usage.
+ *
+ * @author Jerome Gout
+ */
+public class AcceptActionPayloadNodeToolProvider implements INodeToolProvider {
```

Comment:
Unless I'm missing something here, this tool should not be available if it as already been executed (e.g. the same behavior than the New Subject tool on Requirement Def). Could you check that and fix accordingly?

Reply:
I chose to replace the existing element when this tool is called several times.
But you are right, I will change it to be aligned with the New Subject tool.

Reply:
I've added a precondition to remove the tool whenever the payload is already set.